File: tests/fixtures/kotlin_simple/services/UserService.kt
package com.example.services

import com.example.models.User

interface UserService {
    fun findById(id: Int): User?
    fun findAll(): List<User>
    fun save(user: User): User
}

class UserServiceImpl : UserService {
    private val users = mutableListOf<User>()

    override fun findById(id: Int): User? {
        return users.find { it.id == id }
    }

    override fun findAll(): List<User> = users.toList()

    override fun save(user: User): User {
        users.add(user)
        return user
    }

    internal fun count(): Int = users.size
}

fun UserService.exists(id: Int): Boolean {
    return findById(id) != null
}

File: tests/fixtures/csharp_simple/Services/UserService.cs
using System.Collections.Generic;
using MyApp.Models;

namespace MyApp.Services;

public interface IUserService
{
    User FindById(int id);
    List<User> FindAll();
    void Save(User user);
}

public class UserServiceImpl : IUserService
{
    private readonly List<User> _users = new();

    public User FindById(int id)
    {
        return _users[id];
    }

    public List<User> FindAll()
    {
        return _users;
    }

    public void Save(User user)
    {
        _users.Add(user);
    }

    private void Validate(User user)
    {
        if (user == null) throw new ArgumentNullException(nameof(user));
    }
}
